components/launcher/src/server/mod.rs

```diff
@@ -76,21 +85,31 @@ impl Server {
     /// Passing a value of true to the `clean` argument will force the Supervisor to clean the
     /// Launcher's process LOCK before starting. This is useful when restarting a Supervisor
     /// that terminated gracefully.
-    fn init(args: &[String], clean: bool) -> Result<((Receiver, Sender), Child)> {
+    fn init(args: &[String], clean: bool) -> Result<((Receiver, Sender), Child, String)> {
         let (server, pipe) = IpcOneShotServer::new().map_err(Error::OpenPipe)?;
         let supervisor = spawn_supervisor(&pipe, args, clean)?;
         let channel = setup_connection(server)?;
-        Ok((channel, supervisor))
+        Ok((channel, supervisor, pipe))
+    }
+
+    fn remove_pipe(&self) {
+        if fs::remove_file(&self.pipe).is_err() {
+            error!("Could not remove old pipe to supervisor {}", self.pipe);
+        } else {
+            debug!("Removed old pipe to supervisor {}", self.pipe);
+        }
     }
 
     #[allow(unused_must_use)]
     fn reload(&mut self) -> Result<()> {
         self.supervisor.kill();
         self.supervisor.wait();
-        let ((rx, tx), supervisor) = Self::init(&self.args, true)?;
+        let ((rx, tx), supervisor, pipe) = Self::init(&self.args, true)?;
         self.tx = tx;
         self.rx = rx;
         self.supervisor = supervisor;
+        self.remove_pipe();
```

Comment:
A little bit of an explanatory comment about why we're "removing a pipe before saving a pipe" might allay potential confusion when reading this code for the first time.

Reply:
Ooooh, I like that.